CrossAppLogin/Back/src/main/kotlin/com/infomaniak/core/crossapplogin/back/DerivedTokenGeneratorImpl.kt

```diff
@@ -82,14 +83,14 @@ internal class DerivedTokenGeneratorImpl(
         appIntegrityManager.requestClassicIntegrityVerdictToken(dummyChallenge)
         true
     }.cancellable().getOrElse { exception ->
-        if (
-            exception is IntegrityException &&
-            exception.cause?.message?.contains(Regex(INTEGRITY_UNAVAILABLE_ERROR_REGEX)) == true
-        ) {
-            // If we get one of these Integrity exceptions, it means we won't be able to connect to the Service when tying later
-            return@getOrElse false
+        when (exception) {
+            is CustomAppIntegrityException -> false // fDroid
```

Comment:
Maybe this new exception typing could mention that it represents an unsupported configuration like fdroid in this case. Maybe someting like `UnsupportedFdroidIntegrityException` or `UnsupportedIntegrityException` or similar?